Northstar.Client/mod/scripts/vscripts/ui/menu_mod_settings.nut

```diff
@@ -0,0 +1,1111 @@
+untyped
+global function AddModSettingsMenu
+global function AddConVarSetting
+global function AddConVarSettingEnum
+global function AddConVarSettingSlider
+global function AddModTitle
+global function AddModCategory
+global function PureModulo
+
+const int BUTTONS_PER_PAGE = 15
+const string SETTING_ITEM_TEXT = "                        " // this is long enough to be the same size as the textentry field
+
+enum eEmptySpaceType
+{
+	None,
+	TopBar,
+	BottomBar
+}
+
+struct ConVarData {
+	string displayName
+	bool isEnumSetting = false
+	string conVar
+	string type
+
+	string modName
+	string catName
+	bool isCategoryName = false
+	bool isModName = false
+
+	bool isEmptySpace = false
+	int spaceType = 0
+
+	// SLIDER BULLSHIT
+	bool sliderEnabled = false
+	float min = 0.0
+	float max = 1.0
+	float stepSize = 0.05
+	bool forceClamp = false
+
+	bool isCustomButton = false
+	void functionref() onPress
+
+	array< string > values
+	var customMenu
+	bool hasCustomMenu = false
+}
+
+struct {
+	var menu
+	int scrollOffset = 0
+	bool updatingList = false
+
+	array< ConVarData > conVarList
+	// if people use searches - i hate them but it'll do : )
+	array< ConVarData > filteredList
+	string filterText = ""
+	table< int, int > enumRealValues
+	table< string, bool > setFuncs
+	array< var > modPanels
+	array<var> resetModButtons
+	array< MS_Slider > sliders
+	table settingsTable
```

Comment:
are the key/value types known at all? I don't really like table declarations like this tbh

Reply:
This is a remainder of safe i/o - will remove.